### PR TITLE
tools: update ruff configuration to starbase v1

### PR DESCRIPTION
**Pull request by lengau**
Broken into commits for ease of reading.

### CODE REVIEW

File: charmcraft/commands/init.py [hunk 1]
```diff
@@ -71,12 +71,12 @@
     ├── README.md              - Frontpage for your charmhub.io/charm/
     ├── requirements.txt       - PyPI dependencies for your charm, with `ops`
     ├── src
-    │   └── charm.py           - Minimal operator using Python operator framework
+    │   └── charm.py           - Minimal operator using Python operator framework
```
Comment (mr-cal) [inline charmcraft/commands/init.py:74]: Tabs to spaces?  The output still looks OK on the terminal?

**Review by mr-cal — APPROVED** [review]
:+1:

File: charmcraft/commands/init.py (hunk 1 above)
Comment (lengau) [reply charmcraft/commands/init.py:74]: Non-breaking spaces to spaces, and still looks fine in the terminal.

File: tests/test_infra.py [hunk 2]
```diff
@@ -45,12 +45,6 @@ def get_python_filepaths(*, roots=None, python_paths=None):
...
 
 
-@pytest.mark.skipif(os.getenv("RUNNING_TOX"), reason="does not work inside tox")
-def test_pep8():
-    """Verify all files are nicely styled."""
-    pep8_test(get_python_filepaths())
-
-
 def pep8_test(python_filepaths):
     """Helper to check PEP8 (used from this module and from test_init.py to check templates)."""
```
[the remark this answers is not in the document]
Comment (lengau) [reply tests/test_infra.py:49]: That is for another PR, but it's a matter of whether we want to move it or drop the test altogether. We already have ruff, black and codespell in the tox configuration template, so I think I'd rather include running `tox` in a spread test that tests `charmcraft init`

**Review by syu-w — APPROVED** [review]
I thought you wanna do this after the rebase

**Comment by lengau** [comment]
@syu-w I did, but I ended up creating a circular dependency with things I wanted to do and this cut that of.

**Pull request merged**